.doc/README.md

```diff
@@ -0,0 +1,13 @@
+# Design Documentation
+
+This `.doc/` directory houses design documents that are useful for:
```

Comment:
Is `.doc` a convention? We already have `docs` folder for the main docs so I don't think the difference would be clear. If we don't want it in the user facing stuff maybe just `enterpise/docs` for now?

This reminds me of an [ADR](https://github.com/joelparkerhenderson/architecture-decision-record?tab=readme-ov-file#adr-example-templates) doc, which is cool.